roles/solrcloud/tasks/config.yml

```diff
@@ -1,134 +1,88 @@
 ---
-- name: Solrcloud | Check for solr znode
-  ansible.builtin.command: >
-    {{ solr_installation }}/bin/solr zk ls /
-    -z {{ solr_zookeeper_hosts_string }}
-  register: znode_check
-  ignore_errors: true
-  changed_when: false
+# Configure SolrCloud
 
-- name: Solrcloud | Create zk chroot / solr znode using bin/solr
-  ansible.builtin.command: >
-    {{ solr_installation }}/bin/solr zk mkroot {{ solr_znode }}
-    -z {{ solr_zookeeper_hosts_string }}
-  when: solr_znode not in znode_check.stdout_lines and solr_znode != ''
-  changed_when: false
+- name: Solrcloud | Debug Solr variables
+  ansible.builtin.debug:
+    msg:
+      - "solr_installation: {{ solr_installation }}"
+      - "solr_bin: {{ solr_bin }}"
+      - "solr_cloud_download_version: {{ solr_cloud_download_version }}"
+      - "solr_znode: {{ solr_znode | default('undefined') }}"
 
-- name: Solrcloud | Configure SolrCloud init script
-  ansible.builtin.template:
-    src: 'solr.in.sh.j2'
-    dest: '/etc/default/solr.in.sh'
-    owner: '{{ solr_user }}'
-    group: '{{ solr_group }}'
-    mode: "0644"
-    force: true
-  changed_when: false
-  notify: restart SolrCloud
+- name: Solrcloud | Set zookeeper connection string
+  ansible.builtin.set_fact:
+    solr_zookeeper_hosts_string: "{{ solr_zookeeper_hosts | join(',') }}"
+  when: solr_zookeeper_hosts_string is not defined
 
-- name: Solrcloud | Configure high ulimit value for Solr
-  ansible.builtin.template:
-    src: 'solr.conf.j2'
-    dest: '/etc/security/limits.d/solr.conf'
-    mode: "0644"
-    force: true
-  notify: restart SolrCloud
+- name: Solrcloud | Set default solr_znode value based on Solr version
+  ansible.builtin.set_fact:
+    solr_znode: "{% if solr_cloud_download_version is version('9.0.0', '>=') %}solr9{% else %}solr8{% endif %}"
+  when: solr_znode is not defined or solr_znode == ''
 
-- name: Solrcloud | Configure SolrCloud properties
-  ansible.builtin.template:
-    src: 'solr.xml.j2'
-    dest: '{{ solr_data_dir }}/solr.xml'
-    owner: '{{ solr_user }}'
-    group: '{{ solr_group }}'
-    mode: "0644"
-    force: true
-  notify: restart SolrCloud
-  when:
-    - '"8" in solr_cloud_download_version'
+- name: Solrcloud | Check for solr executable
+  ansible.builtin.stat:
+    path: "{{ solr_bin }}"
+  register: solr_binary
 
-- name: Solrcloud | Configure SolrCloud 9 properties
-  ansible.builtin.template:
-    src: 'solr9.xml.j2'
-    dest: '{{ solr_data_dir }}/solr.xml'
-    owner: '{{ solr_user }}'
-    group: '{{ solr_group }}'
-    mode: "0644"
-    force: true
-  notify: restart SolrCloud
-  when:
-    - '"9" in solr_cloud_download_version'
+- name: Solrcloud | Display solr binary status
+  ansible.builtin.debug:
+    var: solr_binary
+  when: not solr_binary.stat.exists
```

Comment:
Um . . . maybe a comment would help here.

Reply:
This is an artifact of debugging.